distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSingleSignOn.java

```diff
@@ -0,0 +1,261 @@
+        } else if (response.getBody().asString()
+                .contains("<title>Redirect</title>")) { // Javascript redirect
+            Pattern pattern = Pattern.compile("encoded *= *\"(.*)\"", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(response.getBody().asString());
+            matcher.find();
+            fullUrl = matcher.group(1);
+        } else {
+            // TODO: this is unexpected. Log an error?
+        }
+        return fullUrl;
+    }
+
+    private String parseUrl(Response response) {
+        String url = getRedirectUrl(response);
+
+        // We want just the base url, not all of the parameters
+        if (url.contains("?")) {
+            url = url.substring(0, url.indexOf('?'));
+        }
+        return url;
+    }
+
+    private Map<String, String> parseParams(Response response) throws Exception {
+        String url = getRedirectUrl(response);
+
+        // RestAssured expects url query parameters to be in the form of a Dictionary.
+        List<NameValuePair> paramList = URLEncodedUtils.parse(new URI(url), "UTF-8");
+        HashMap<String, String> jsonParams = new HashMap<>();
+        for (NameValuePair param : paramList) {
+            jsonParams.put(param.getName(), param.getValue());
+        }
+        return jsonParams;
+    }
+
+    @Test
+    public void testBadUsernamePassword() throws Exception {
+
+        // Request search service
+        Response searchResponse = given().
+                redirects().follow(false).
+
+                expect().
+                statusCode(302).
+
+                when().
+                get(SEARCH_URL.getUrl());
+
+        // Pass bad credentials to IDP
+        Response IDPResponse = given().auth().preemptive().basic("definitely", "notright")
+                .param("AuthMethod", "up").params(parseParams(searchResponse))
+
+                // TODO: This should be a 401, not 500
+                .expect().statusCode(500).body(containsString("could not be authenticated"))
+
+                .when().get(parseUrl(searchResponse) + "/sso");
+    }
+
+    private void checkSaml(Map<String, String> samlParams) throws Exception {
+        String samlResponse = RestSecurity.inflateBase64(samlParams.get("SAMLResponse"));
+
+        assertThat(samlParams.get("SigAlg"), not(isEmptyOrNullString()));
+        assertThat(samlParams.get("Signature"), not(isEmptyOrNullString()));
+
+        assertThat(samlResponse, allOf(containsString("urn:oasis:names:tc:SAML:2.0:status:Success"),
+                containsString("ds:SignatureValue"), containsString("saml2:Assertion")));
+        assertThat(samlParams.get("RelayState").length(),
+                is(both(greaterThan(0)).and(lessThanOrEqualTo(80))));
+    }
+
+    @Test
+    public void testIdpAuth() throws Exception {
+
+        Response initialWhoamiResponse = given().
+                expect().
+                statusCode(200).
+                body(not(containsString("admin"))).
```

Comment:
Should use an xpath and a positive instead of a not test since you know what you are expecting.